flowmachine/flowmachine/core/query.py

```diff
@@ -503,20 +503,21 @@ def to_sql(
 
         def do_query() -> Query:
             logger.debug("Getting storage lock.")
-            with rlock(self.redis, self.md5):
+            q_state_machine = QueryStateMachine(self.redis, self.md5)
```

Comment:
There is a lot of repetition between the `do_query()` methods in the `ModelResult` and `Query` classes. Is this an indication that this functionality should live somewhere else where it can be shared?